morgoth/base/static/js/components/AddonForm.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { PropTypes as pt } from 'react';
+import { browserHistory } from 'react-router';
+import { reduxForm } from 'redux-form';
+
+import FlatButton from 'material-ui/FlatButton';
+import LinearProgress from 'material-ui/LinearProgress';
+import RaisedButton from 'material-ui/RaisedButton';
+import NavigationChevronLeft from 'material-ui/svg-icons/navigation/chevron-left';
+import TextField from 'material-ui/TextField';
+import { Toolbar, ToolbarGroup, ToolbarSeparator } from 'material-ui/Toolbar';
+
+import ErrorSnackbar from './stateless/ErrorSnackbar.jsx';
+import FetchErrorList from './stateless/FetchErrorList.jsx';
+import LoadingIndicator from './stateless/LoadingIndicator.jsx';
+
+
+const style = {
+  toolbar: {
+    justifyContent: 'flex-end',
+  },
+  refresh: {
+    display: 'inline-block',
+    position: 'relative',
+  },
+};
```

Comment:
ahhhh this is so weird to me still but everyone seems to enjoy doing it these days